Library/ViewModels/DashboardViewModelTests.swift

```diff
@@ -335,4 +335,68 @@ internal final class DashboardViewModelTests: TestCase {
       self.vm.inputs.showHideProjectsDrawer()
     }
   }
+
+  func testTrackingEvents_CreatorDashboardViewed() {
+    let projects = (0...4).map { .template |> Project.lens.id .~ $0 }
+
+    withEnvironment(apiService: MockService(fetchProjectsResponse: projects)) {
+      XCTAssertEqual([], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      XCTAssertEqual(["Page Viewed"], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillDisappear()
+
+      XCTAssertEqual(["Page Viewed"], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      XCTAssertEqual(["Page Viewed", "Page Viewed"], self.segmentTrackingClient.events)
+    }
+  }
+
+  func testTrackingEvents_CreatorDashboardSwitchProjectClicked() {
+    let project1 = Project.template
+    let project2 = .template |> Project.lens.id .~ 4
+    let projects = [project1, project2]
+
+    withEnvironment(apiService: MockService(fetchProjectsResponse: projects)) {
+      XCTAssertEqual([], self.segmentTrackingClient.events)
+
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      self.scheduler.advance()
```

Comment:
Do you need this `advance` noticed above test wasn't using it.

Reply:
yes actually. it needs time for the `fetchProjectsResponse` to complete and set the given projects I believe. these tests fail without it.